python3.12-jax: PJitErrorTest::testAxisResourcesMismatch test fails on aarch64-darwin

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

Hi,

There is a failing test on `python3.12-jax` on Unstable, at least on my machine (`aarch64-darwin`).
The error is the same as the one for which other tests have already been disabled on `darwin` as I have seen in the package.
The test in question is `PJitErrorTest::testAxisResourcesMismatch`.
Adding
```
"--deselect tests/pjit_test.py::PJitErrorTest::testAxisResourcesMismatch"
```
to the `pytestFlagsArray` Darwin optionals fixes it for me, obviously. Maybe it should be added in the package on Nixpkgs.

### Steps to reproduce

- Build `python3.12-jax` from Unstable on `aarch64-darwin`
-  1 test fails (`PJitErrorTest::testAxisResourcesMismatch`)

### Expected behaviour

Package passes tests

### Screenshots

_No response_

### Relevant log output

```console
python3.12-jax> =========================== short test summary info ============================
python3.12-jax> FAILED tests/pjit_test.py::PJitErrorTest::testAxisResourcesMismatch - SystemError: nanobind::detail::nb_func_error_except(): exception could not ...
python3.12-jax> ========= 1 failed, 24585 passed, 13082 skipped in 1740.48s (0:29:00) ==========
error: builder for '/nix/store/1qwahz31gys8wr1l3ikymg3r34wpv4ys-python3.12-jax-0.5.0.drv' failed with exit code 1;
       last 10 log lines:
       >                  is_leaf: Callable[[Any], bool] | None = None) -> Any:
       >       """Alias of :func:`jax.tree.map`."""
       >       leaves, treedef = tree_flatten(tree, is_leaf)
       > >     all_leaves = [leaves] + [treedef.flatten_up_to(r) for r in rest]
       > E     SystemError: nanobind::detail::nb_func_error_except(): exception could not be translated!
       >
       > jax/_src/tree_util.py:357: SystemError
       > =========================== short test summary info ============================
       > FAILED tests/pjit_test.py::PJitErrorTest::testAxisResourcesMismatch - SystemError: nanobind::detail::nb_func_error_except(): exception could not ...
       > ========= 1 failed, 24585 passed, 13082 skipped in 1740.48s (0:29:00) ==========
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.19.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

### Are you using nix-darwin?

No, I am not using nix-darwin.

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)
@samuela 

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+darwin%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/jax/default.nix
+++ b/pkgs/development/python-modules/jax/default.nix
@@ -95,6 +95,8 @@ buildPythonPackage rec {
     ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [
       # SystemError: nanobind::detail::nb_func_error_except(): exception could not be translated!
+      # reported at: https://github.com/jax-ml/jax/issues/26106
+      "--deselect tests/pjit_test.py::PJitErrorTest::testAxisResourcesMismatch"
       "--deselect tests/shape_poly_test.py::ShapePolyTest"
       "--deselect tests/tree_util_test.py::TreeTest"
     ];
@@ -150,6 +152,7 @@ buildPythonPackage rec {
     ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [
       # SystemError: nanobind::detail::nb_func_error_except(): exception could not be translated!
+      # reported at: https://github.com/jax-ml/jax/issues/26106
       "testInAxesPyTreePrefixMismatchError"
       "testInAxesPyTreePrefixMismatchErrorKwargs"
       "testOutAxesPyTreePrefixMismatchError"

